graylog2-web-interface/src/views/logic/views/ViewTransformer.test.js

```diff
@@ -0,0 +1,129 @@
+
+    it('should add the timerange to the widget', () => {
+      const query = Query.builder()
+        .id('query-id')
+        .timerange({ type: 'relative', range: 365 })
+        .build();
+
+      const search = Search.builder()
+        .id('search-id')
+        .queries([query])
+        .build();
+
+      const viewState: ViewState = ViewStateGenerator(View.Type.Search);
+
+      const viewStateMap: ViewStateMap = Map.of('query-id', viewState);
+      const searchView = View.builder()
+        .type(View.Type.Search)
+        .state(viewStateMap)
+        .search(search)
+        .build();
+      const dashboardView = viewTransformer(searchView);
+      expect(dashboardView.state.get('query-id').widgets.first().timerange).toBe(query.timerange);
+      expect(dashboardView.state.get('query-id').widgets.first().query).toBeUndefined();
+      expect(dashboardView.state.get('query-id').widgets.first().streams).toStrictEqual([]);
+    });
+
+    it('should add the query to the widget', () => {
+      const query = Query.builder()
+        .id('query-id')
+        .query({ type: 'elasticsearch', query_string: 'author: "Karl Marx"' })
+        .build();
+
+      const search = Search.builder()
+        .id('search-id')
+        .queries([query])
+        .build();
+
+      const viewState: ViewState = ViewStateGenerator(View.Type.Search);
+
+      const viewStateMap: ViewStateMap = Map.of('query-id', viewState);
+      const searchView = View.builder()
+        .type(View.Type.Search)
+        .state(viewStateMap)
+        .search(search)
+        .build();
+      const dashboardView = viewTransformer(searchView);
+      expect(dashboardView.state.get('query-id').widgets.first().timerange).toBeUndefined();
+      expect(dashboardView.state.get('query-id').widgets.first().query).toBe(query.query);
+      expect(dashboardView.state.get('query-id').widgets.first().streams).toStrictEqual([]);
+    });
+
+    it('should add the streams to the widget', () => {
+      const query = Query.builder()
+        .id('query-id')
+        .filter({ type: 'or', filters: List.of(Map.of('type', 'stream', 'id', '1234-abcd')) })
+        .build();
+
+      const search = Search.builder()
+        .id('search-id')
+        .queries([query])
+        .build();
+
+      const viewState: ViewState = ViewStateGenerator(View.Type.Search);
+
+      const viewStateMap: ViewStateMap = Map.of('query-id', viewState);
+      const searchView = View.builder()
+        .type(View.Type.Search)
+        .state(viewStateMap)
+        .search(search)
+        .build();
+      const dashboardView = viewTransformer(searchView);
+      expect(dashboardView.state.get('query-id').widgets.first().timerange).toBeUndefined();
+      expect(dashboardView.state.get('query-id').widgets.first().query).toBeUndefined();
+      expect(dashboardView.state.get('query-id').widgets.first().streams).toStrictEqual(['1234-abcd']);
+    });
+  });
+
+  describe('transform with the whole view', () => {
```

Comment:
The test description is a bit too vague for me. What does the block actually do?